africanus/calibration/phase_only/phase_only.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+from africanus.util.docs import DocstringTemplate
+from africanus.calibration.utils import residual_vis, check_type
+from africanus.util.numba import generated_jit, njit
+
+DIAG_DIAG = 0
+DIAG = 1
+FULL = 2
+
+
+def jacobian_factory(mode):
+    if mode == DIAG_DIAG:
+        def jacobian(a1j, blj, a2j, sign, out):
+            out[...] = sign * a1j * blj * a2j.conjugate()
+            # for c in range(out.shape[-1]):
+            #     out[c] = sign * a1j[c] * blj[c] * a2j[c].conjugate()
+    elif mode == DIAG:
+        def jacobian(a1j, blj, a2j, sign, out):
+            out[...] = 0
+    elif mode == FULL:
+        def jacobian(a1j, blj, a2j, sign, out):
+            out[...] = 0
+    return njit(nogil=True)(jacobian)
```

Comment:
This may now be a candidate for the following:

```suggestion
    return njit(nogil=True, inline='always')(jacobian)
```

Reply:
Will add. I actually did that and removed it while I was trying to figure out why the one test was failing